### PR TITLE
FlxMouseEventManager: add move, click and wheel events, closes #2085

### LINKED ISSUE #2085: Proper CLICK event (and more?)
`FlxMouseEventManager` handles `DOWN`, `UP`, `OVER`, and `OUT`, but not `CLICK` and its variants or `MOVE`. Should it have callbacks for these others too? `CLICK` is probably a `DOWN` and `UP` on the same object, and `MOVE` is maybe `OVER` + move.

Then what about all the variants of `MIDDLE` and `RIGHT` having callbacks? I personally think no.

Main use case is haxeui expecting `CLICK`, `MOVE`, and `WHEEL` to be handled. I've been overloading `UP`'s handler with `CLICK`, but I'm not sure how good of a solution that is in general.

### PR DESCRIPTION
**Pull request by MSGhero**
Closes #2085 

Added `MOVE`, `CLICK`, and `WHEEL` mouse callbacks. It works, but I haven't extensively tested it yet, which I'll be doing with HaxeUI and my game. MMB/RMB click callbacks don't seem as necessary, and I decided not to `DOUBLE_CLICK` unless someone really wants it.

Also added `justMoved` to `FlxMouse`. It seemed more appropriate there than in `FlxPointer` since pointer doesn't have a once-per-frame `update()`.

AS3's mouse events  aren't tied to the framerate in my testing, and flixel can't do anything to match that, but that's ok.

### CODE REVIEW

**Comment by MSGhero** [comment]
Not seeing any issues.

**Comment by MSGhero** [comment]
I suddenly need double clicking in my game, so I'll be adding that.

**Comment by MSGhero** [comment]
Looks like there's an issue with dev haxe in the CI builds, unrelated to flixel.

File: flixel/input/mouse/FlxMouseEventManager.hx [hunk 1]
```diff
@@ -36,10 +36,20 @@ class FlxMouseEventManager extends FlxBasic
...
+	
+	private static var _mouseClickedTime:Int = -1;
 
 	private static var _point:FlxPoint = FlxPoint.get();
 	
 	/**
+	 * The maximum amount of time between two clicks that is considered a double click, in milliseconds.
+	 * @since 4.4.0
+	 */
+	public static var MAX_DOUBLE_CLICK_DELAY:Int = 500;
```
Comment (Gama11) [inline flixel/input/mouse/FlxMouseEventManager.hx]: Hm.. the `UPPER_CASE` naming implies to me that this is a constant (declared `inline` for instance) that can't be changed.

**Comment by Gama11** [comment]
Presumably this can be tested using a HaxeUI demo?

**Comment by MSGhero** [comment]
Yeah, but I'm (still) waiting for them to resolve an incompatibility. Right now you have to comment out a line in the core UI library for it to work with flixel.

**Comment by MSGhero** [comment]
Last time I added an external lib...
image
image
Seems like it's easier to do from demos at least. I'll make one and leave it WIP so I don't forget, but it won't work or pass CI until haxeui/haxeui-core#140 is resolved.

**Comment by Gama11** [comment]
Hm? You want to add haxeui to Flixel's CI?

**Comment by MSGhero** [comment]
Don't I need to add it to flixel-demo's CI so the UI demo can be part of the check?

**Comment by Gama11** [comment]
I'm not sure what you're trying to do to begin with. :) What UI demo?

**Comment by MSGhero** [comment]
> Presumably this can be tested using a HaxeUI demo?

Oh, did you not mean a flixel-demos demo?

**Comment by Gama11** [comment]
No, that's why I said HaxeUI demo. :D

**Comment by MSGhero** [comment]
Lol... I tested this extensively just with flixel, and I've been using it with HaxeUI in my game without any problems. The UI lib isn't developed enough where I have demos and CI yet, but I could add one in the future.

**Comment by Gama11** [comment]
But HaxeUI has demos, right? Stuff like http://haxeui.org:1234/demos/explorer/. Does that just not work with haxeui-flixel yet?

**Comment by MSGhero** [comment]
All of the applicable parts of that demo (ie not the native UI elements) work fine from the flixel backend. 

I meant flixel-specific demos, like setting up an atlas to get sprites from, which is something none of the other backends do.

**Comment by Gama11** [comment]
Wow, really? I didn't expect it to be this far along, I'll have to check that out.

**Comment by MSGhero** [comment]
If you do try it out, just note that I have click, move, and wheel events disabled, so you may have weird behavior if the component is expecting one of those.

**Comment by MSGhero** [comment]
Fixed an edge case with hitting breakpoints or otherwise tabbing out of the window while the mouse was down.

**Comment by Gama11** [comment]
Thanks, these are very useful additions. :)

**Pull request merged**